### DISCUSSION TITLE
Using Termux to write html documents

### DISCUSSION THREAD

**Discussion by JoshuaDK-stack** [community/community#42080] [Programming Help]
Hello good afternon, Ive been trying to create a webpage that would store data from an html form using termux. So far Ive managed to run the javascript code on termux using vim. The only problem I have is that google can not access the html file if i create it termux. 

I can create a new text file and run node.js code to write data from an html form inside the termux terminal using vim. I can only view it from the terminal.

This website would act as a server from what ive been trying to do. This is mainly for testing purposes i understand that if i wanted to store more data i would need to migrate to a pc. For now i just want to test it.

The idea is that the user would enter their login details via my webpage node.js would create a textfile on my phone and store the data. 

I cant run the html on google. Google can not find it.

The prodject uses HTML, CSS, Javascript, Node.js, termux as the emulator and vim as the editor. 

I cant find any information online on how to give access to google in order to run the html file stored in termux. I found how to run google CLI but that is not what i want. I want to store everything locally and not on a server. I want google to load my page but data must be stored within my phone. Or later on my pc. 

Im still new to this, i can explain in more detail if anyone is willing to help. I have pictures and the code and i can share the steps i took to get here.

So far the only thing i can come up with is chmod 777 com.termux 

It still doesnt allow me to open the html file via google.

Potential idea to include a pkg that would allow me to root my phone without a pc

**Comment by Asyrofimood** [comment]
~ $ pkg upgrade && pkg upgrade
Checking availability of current mirror:
[*] https://packages-cf.termux.dev/apt/termux-main/: ok
Hit:1 https://packages-cf.termux.dev/apt/termux-main stable InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Checking availability of current mirror:
[*] https://packages-cf.termux.dev/apt/termux-main/: ok
Hit:1 https://packages-cf.termux.dev/apt/termux-main stable InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
~ $ pkg install git
Checking availability of current mirror:
[*] https://packages-cf.termux.dev/apt/termux-main/: ok
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
git is already the newest version (2.55.0).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
~ $ git clone https://github.com/him0x/otp
Cloning into 'otp'...
remote: Enumerating objects: 28, done.
remote: Counting objects: 100% (28/28), done.
remote: Compressing objects: 100% (22/22), done.
remote: Total 28 (delta 7), reused 22 (delta 3), pack-reused 0 (from 0)
Receiving objects: 100% (28/28), 49.32 KiB | 1.12 MiB/s, done.
Resolving deltas: 100% (7/7), done.
~ $ lw
No command lw found, did you mean:
 Command ln in package coreutils
 Command lp in package cups
 Command gw in package gw from the x11-repo repository
 Command iw in package iw from the root-repo repository
 Command lf in package lf
 Command ld in package lld
 Command lr in package lr
 Command lx in package lux-cli
 Command lc in package mono
 Command lz in package mtools
~ $ lw
No command lw found, did you mean:
 Command ln in package coreutils
 Command lp in package cups
 Command gw in package gw from the x11-repo repository
 Command iw in package iw from the root-repo repository
 Command lf in package lf
 Command ld in package lld
 Command lr in package lr
 Command lx in package lux-cli
 Command lc in package mono
 Command lz in package mtools
~ $ ls
otp  storage
~ $ cd otp
~/otp $ pkg install python
Checking availability of current mirror:
[*] https://packages-cf.termux.dev/apt/termux-main/: ok
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
python is already the newest version (3.14.6-1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
~/otp $ termux-setup-storage

It appears that directory '~/storage' already exists.
This script is going to rebuild its structure from
scratch, wiping all dangling files. The actual storage
content IS NOT going to be deleted.

Do you want to continue? (y/n) y
~/otp $ pip install -r requirements.txt
Collecting requests>=2.25.0 (from -r requirements.txt (line 1))
  Downloading requests-2.34.2-py3-none-any.whl.metadata (4.8 kB)
Collecting colorama>=0.4.0 (from -r requirements.txt (line 2))
  Downloading colorama-0.4.6-py2.py3-none-any.whl.metadata (17 kB)
Collecting charset_normalizer<4,>=2 (from requests>=2.25.0->-r requirem
ents.txt (line 1))
  Downloading charset_normalizer-3.4.9-py3-none-any.whl.metadata (41 kB
)
Collecting idna<4,>=2.5 (from requests>=2.25.0->-r requirements.txt (li
ne 1))
~ python main.py

**Comment by v-coder247** [comment]
Welcome to the GitHub Community, @JoshuaDK-stack! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!
